### PR TITLE
Feature/multiple version support

### PR DESCRIPTION
**Pull request by johandahlberg**
Ok - first of all. This is not as bad as it looks. Most of the changes are just files that have been moved around.

This introduces a number of changes to the way Piper is built. Most importantly it introduces the possibility to build a deployable zip file using:

    sbt universal:packageBin

See `README.md` for detailed instructions.

Another important change is that the git tags are now used for versioning - to to create a new version one should create a tag. Builds which are not created from a commit which is tagged and in a clean working directory will be marked according to: `<base-tag>-<nbr of commits on top of tag>-<short sha-hash>`. 

Please not that the tests are not up and running again, but I'm saving that for a later PR to keep the size of this one manageable.

There is still some testing to be done on this - but I wanted some feedback none the less. Don't merge until I confirm all is :ok_hand:.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,23 +21,30 @@ All supported workflows are available in the `workflows` directory in the projec
...
-    git clone https://github.com/Molmed/piper.git
+    git clone https://github.com/NationalGenomicsInfrastructure/piper.git
     cd piper
-    ./setup.sh <path to install Piper to. Default is: $HOME/Bin/Piper>
-    # Follow the instructions printed by the setup script.
+    sbt universal:packageBin
+    # This will create two zip files.
+    # Unzip these where you wish to install piper
+    ./piper/target/universal/piper-v[version].zip
+    ./setup-file-creator/target/universal/setupfilecreator-v[version].zip
```
Comment (b97pla) [inline README.md]: Are the zip-file extraction interactive or is it necessary to specify a location on the command line?

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,6 @@
 language: scala
 scala:
-   - "2.9.2"
+   - "2.10.1"
```
Comment (b97pla) [inline .travis.yml]: version in `build.sbt` is `2.10.3`, not sure if that matters

File: piper/src/main/resources/workflows/Exome.sh [hunk 3]
```diff
@@ -83,7 +87,7 @@ fi
 #---------------------------------------------
 # Run the qscript
 #---------------------------------------------
-piper -S ${SCRIPTS_DIR}/DNABestPracticeVariantCalling.scala \
+piper -S ${_THIS_SCRIPT_LOCATION}/../qscripts/DNABestPracticeVariantCalling.scala \
 	      --xml_input ${PIPELINE_SETUP} \
 	      --isExome \
 	      --gatk_interval_file ${INTERVALS} \
```
Comment (b97pla) [inline piper/src/main/resources/workflows/Exome.sh]: to be consistent with the other workflows, should it be `--global_config ${_THIS_SCRIPT_LOCATION}/../conf/uppmax_global_config.xml` below?

File: README.md (hunk 1 above)
Comment (johandahlberg) [inline README.md]: It will unzip in the same directory where the file is located.

**Comment by b97pla** [comment]
Looks good to me, let me know when you are :ok_hand:

File: .travis.yml (hunk 2 above)
Comment (johandahlberg) [inline .travis.yml]: Good point. Shouldn't matter to much for a minor version, but I change it nonetheless.

File: piper/src/main/resources/workflows/Exome.sh (hunk 3 above)
Comment (johandahlberg) [inline piper/src/main/resources/workflows/Exome.sh]: It should indeed.

**Comment by johandahlberg** [comment]
I've updated to match your comments, and I've also tested on Uppmax. I've not run a full test - but I have confirmed everything starts and loads the correct config files, etc. So there is no reason not expect it shouldn't work.

This should be ready to merge now.

**Comment by b97pla** [comment]
Nice work, great functionality! :dancers:

**Pull request merged**